src/event_trigger.c

```diff
@@ -109,42 +110,53 @@ extract_addrnames(ArrayType *arr)
 }
 
 static EventTriggerDropTableConstraint *
-make_event_trigger_drop_table_constraint(char *constraint_name, char *schema, char *table)
+make_event_trigger_drop_table_constraint(const char *constraint_name, const char *schema,
+										 const char *table)
 {
 	EventTriggerDropTableConstraint *obj = palloc(sizeof(EventTriggerDropTableConstraint));
 
 	*obj =
-		(EventTriggerDropTableConstraint){ .obj = { .type = EVENT_TRIGGER_DROP_TABLE_CONSTRAINT },
-										   .constraint_name = constraint_name,
-										   .schema = schema,
-										   .table = table };
+		(EventTriggerDropTableConstraint){
+		.obj = {
+			.type = EVENT_TRIGGER_DROP_TABLE_CONSTRAINT,
+		},
+		.constraint_name = constraint_name,
+		.schema = schema,
+		.table = table,
+	};
 
 	return obj;
 }
 
-static EventTriggerDropIndex *
-make_event_trigger_drop_index(char *index_name, char *schema)
+static EventTriggerDropRelation *
+make_event_trigger_drop_index(const char *index_name, const char *schema)
 {
-	EventTriggerDropIndex *obj = palloc(sizeof(EventTriggerDropIndex));
+	EventTriggerDropRelation *obj = palloc(sizeof(EventTriggerDropRelation));
 
-	*obj = (EventTriggerDropIndex){
-		.obj = { .type = EVENT_TRIGGER_DROP_INDEX },
-		.index_name = index_name,
+	*obj = (EventTriggerDropRelation){
+		.obj = {
+			.type = EVENT_TRIGGER_DROP_INDEX,
+		},
+		.name = index_name,
 		.schema = schema,
 	};
+
 	return obj;
 }
 
-static EventTriggerDropTable *
-make_event_trigger_drop_table(char *table_name, char *schema)
+static EventTriggerDropRelation *
+make_event_trigger_drop_table(const char *table_name, const char *schema, char relkind)
 {
-	EventTriggerDropTable *obj = palloc(sizeof(EventTriggerDropTable));
+	EventTriggerDropRelation *obj = palloc(sizeof(EventTriggerDropRelation));
```

Comment:
Will it be better to call `palloc0`?
```suggestion
	EventTriggerDropRelation *obj = palloc0(sizeof(EventTriggerDropRelation));
```

Reply:
The current approach is consistent with other functions in this file and the object is, AFAIK, fully filled in and initialized. I'll leave this for a refactor of the whole file in case this is important.

Reply:
You are right and it was originally so.